src/widgets/helper/ExpandableLayout.cpp

```diff
@@ -0,0 +1,69 @@
+#include "widgets/helper/ExpandableLayout.hpp"
+
+#include "widgets/Label.hpp"
+
+namespace chatterino {
+
+ExpandableLayout::ExpandableLayout(const QString &title, QWidget *parent)
+    : BaseWidget(parent)
+{
+    auto layout = new QVBoxLayout(this);
+    layout->setContentsMargins(0, 0, 0, 0);
+
+    this->toggleButton_ = new QToolButton(this);
+    this->content_ = new QWidget(this);
+
+    this->toggleButton_->setCheckable(true);
+    this->toggleButton_->setText(title);
+    this->toggleButton_->setArrowType(Qt::ArrowType::DownArrow);
+    this->toggleButton_->setToolButtonStyle(Qt::ToolButtonTextBesideIcon);
+    this->toggleButton_->setStyleSheet("QToolButton {border: none;}");
+
+    QObject::connect(this->toggleButton_, &QToolButton::toggled, this,
+                     &ExpandableLayout::toggle);
+    // Forward QToolButton::toggled signal
+    QObject::connect(this->toggleButton_, &QToolButton::toggled,
+                     [this](bool toggled) { emit buttonToggled(toggled); });
+
+    layout->addWidget(this->toggleButton_);
+    layout->setAlignment(this->toggleButton_, Qt::AlignHCenter);
+
+    layout->addWidget(this->content_);
+
+    this->setLayout(layout);
+
+    // Collapsed per default
+    this->toggle(false);
+}
+
+void ExpandableLayout::toggle(bool expand)
+{
+    if (expand)
+    {
+        this->toggleButton_->setArrowType(Qt::ArrowType::UpArrow);
+        this->content_->show();
+    }
+    else
+    {
+        // Collapse
+        this->toggleButton_->setArrowType(Qt::ArrowType::DownArrow);
+        this->content_->hide();
+    }
```

Comment:
```suggestion
    this->toggleButton_->setArrowType(expand ? Qt::ArrowType::UpArrow
                                             : Qt::ArrowType::DownArrow);
    this->content_->setVisible(expand);
```

Reply:
Thanks, I applied your suggestion in 0dd6170 via a rebase.